cinema/views.py

```diff
@@ -118,12 +142,17 @@ class OrderPagination(PageNumberPagination):
     max_page_size = 100
 
 
-class OrderViewSet(viewsets.ModelViewSet):
+class OrderViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    viewsets.GenericViewSet
+):
     queryset = Order.objects.prefetch_related(
         "tickets__movie_session__movie", "tickets__movie_session__cinema_hall"
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    permission_classes = (IsAuthenticated,)
```

Comment:
The `OrderViewSet` is missing the custom permission class `IsAdminOrIfAuthenticatedReadOnly`. This is crucial for ensuring the correct access control as per the task requirements. Please add it to the `permission_classes`.